### DISCUSSION TITLE
Zero install?

### DISCUSSION THREAD

**Discussion by silouanwright** [pnpm/pnpm#5698] [Ideas]
One of the great features of Yarn, is Plug N Play and Zero installs. Is there an idea if this will ever be coming to `pnpm`?

https://yarnpkg.com/features/zero-installs

**Comment by iulspop** [comment]
Any comments on this from the maintainers? Is this anywhere on the roadmap?

**Reply by zkochan** [reply]
No, I don't think we will ever work on such feature. One of the points of using pnpm is reducing the amount of disk space used by node_modules. If we will commit packages to the repository, pnpm won't be able to link them from a global store and reduce disk space usage.

**Comment by SleeplessByte** [comment]
You can simulate the behaviour by more or less opting out of pnpm altogether (for example because you want to set-up a docker image that doesn't allow you to write during run).

```ini
# .npmrc
node-linker=hoisted
symlink=false
```

Now `corepack pnpm install --offline --frozen-lockfile` yields no resolution step and nothing pulled from any network.

That said, you usually probably don't want this at all ;)
